cli.py

```diff
@@ -67,21 +75,62 @@ def main(log_level):
+              type=str, help="D3A websocket URI")
+@click.option('-i', '--simulation-config-path', type=str,
+              help="Path to simulation config file (accept absolute and relative path)")
+@click.option('--run-on-redis', is_flag=True, default=False,
+              help="Start the client using the Redis API")
+def run(base_setup_path, setup_module_name, username, password, domain_name, web_socket,
+        simulation_config_path, run_on_redis, **kwargs):
     if username is not None:
         os.environ["API_CLIENT_USERNAME"] = username
     if password is not None:
         os.environ["API_CLIENT_PASSWORD"] = password
     os.environ["API_CLIENT_DOMAIN_NAME"] = domain_name
     os.environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"] = web_socket
     os.environ["API_CLIENT_RUN_ON_REDIS"] = "true" if run_on_redis else "false"
+    set_simulation_file_env(base_setup_path, simulation_config_path, run_on_redis)
+
+    load_client_script(base_setup_path, setup_module_name)
+
 
+def load_client_script(base_setup_path, setup_module_name):
     try:
-        importlib.import_module(f"d3a_api_client.setups.{setup_module_name}")
+        if base_setup_path is None:
+            importlib.import_module(f"d3a_api_client.setups.{setup_module_name}")
+        elif base_setup_path and os.path.isabs(base_setup_path):
+            sys.path.append(base_setup_path)
+            importlib.import_module(setup_module_name)
+        else:
+            setup_file_path = os.path.join(os.getcwd(), base_setup_path)
+            sys.path.append(setup_file_path)
+            importlib.import_module(setup_module_name)
 
     except D3AException as ex:
         raise click.BadOptionUsage(ex.args[0])
     except ModuleNotFoundError as ex:
         log.error("Could not find the specified module")
+
+
+def set_simulation_file_env(base_setup_path, simulation_config_path, run_on_redis):
+    if run_on_redis is True:
+        os.environ["SIMULATION_CONFIG_FILE_PATH"] = ""
+        return
+    if simulation_config_path is None:
+        raise ApiSimulationConfigException(f"simulation_config_path must be provided")
+    elif os.path.isabs(simulation_config_path):
+        os.environ["SIMULATION_CONFIG_FILE_PATH"] = simulation_config_path
+    elif base_setup_path is None:
+        os.environ["SIMULATION_CONFIG_FILE_PATH"] = \
+            os.path.join(api_client_path, 'setups', simulation_config_path)
+    elif base_setup_path is not None:
+        if os.path.isabs(base_setup_path):
+            os.environ["SIMULATION_CONFIG_FILE_PATH"] = \
+                os.path.join(base_setup_path, simulation_config_path)
+        else:
+            os.environ["SIMULATION_CONFIG_FILE_PATH"] = \
+                os.path.join(os.getcwd(), base_setup_path, simulation_config_path)
+
+    else:
+        os.environ["SIMULATION_CONFIG_FILE_PATH"] = \
+            os.path.join(os.getcwd(), simulation_config_path)
```

Comment:
I personally prefer methods that return values rather than changing dictionaries in place (I think we talked about it a few days ago on slack). It would be good to create a method that just creates the correct path, and subsequently assign the returned value to the `os.environ['SIMULATION_CONFIG_FILE_PATH']` variable. 
For example:
```python
if run_on_redis:
    os.environ['SIMULATION_CONFIG_FILE_PATH'] = None
else:
    os.environ['SIMULATION_CONFIG_FILE_PATH'] = create_simulation_config_path(
        base_setup_path, simulation_config_path)
```
and the method might look something like:
```python
def create_simulation_config_path(base_setup_path, simulation_config_path):
    # For example
    if os.path.isabs(simulation_config_path):
        return simulation_config_path
    # and so on
```